impl/src/planning.rs

```diff
@@ -194,8 +195,16 @@ impl CrateCatalogEntry {
    *
    * Not for use except during planning as path is local to run location.
    */
-  pub fn expected_vendored_path(&self) -> String {
-    format!("./{}{}", VENDOR_DIR, &self.package_ident)
+  pub fn expected_vendored_path(&self, workspace_path: &String) -> String {
+    let mut dir = find_workspace_root(None).unwrap_or(PathBuf::from("."));
+    dir.push(workspace_path.trim_start_matches('/'));
+
+    format!(
+      "{}/{}{}",
+      dir.display().to_string(),
+      VENDOR_DIR,
+      &self.package_ident
+    )
```

Comment:
Update: Looking at the usage of `expected_vendored_path` it looks like it's just passed to `std::fs::metadata`, so it can probably just return std::path::Path, or std::path::PathBuf.

(There may be other invocations that I'm missing though)

Reply:
I've updated this to use `Pathbuf::join` but didn't change the return type.